core/sequencer/sequencer.go

```diff
@@ -95,50 +95,51 @@ func (s *Sequencer) AddDirectory(dirIDs ...string) {
 	}
 }
 
-// RunBatchForAllMasterships runs RunBatch on all directires this sequencer is currently master for.
-func (s *Sequencer) RunBatchForAllMasterships(ctx context.Context, batchSize int32) error {
-	glog.Infof("RunBatchForAllMasterships")
+// ForAllMasterships runs f once for all directories this server is master for.
+func (s *Sequencer) ForAllMasterships(ctx context.Context, f func(ctx context.Context, dirID string) error) error {
 	cctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 	masterships, err := s.tracker.Masterships(cctx)
 	if err != nil {
 		return err
 	}
-
 	var lastErr error
-	for dirID, whileMaster := range masterships {
+	for dirID, mastershipCtx := range masterships {
+		if err := f(mastershipCtx, dirID); err != nil {
+			lastErr = err
```

Comment:
Future improvement: It'd be nice if all errors except the last weren't discarded. Consider aggregating them. As is, the last error is likely to end up getting logged twice, potentially implying it occurred twice (since you already log all errors that occur in `f` on line 126).

Reply:
done